docs/dev/components/components.md

```diff
@@ -0,0 +1,94 @@
+# Components/Maven Modules
+
+Gaffer has lots of components, which can be split into different categories. This section of the developer doc provides more detail on the most important components.
+
+## Key components
+
+For more developer information on these key components, see their associated page.
+
+- [Operation](operation.md): Classes for the Operation interfaces and core operation implementations.
+- [Cache](cache.md): Classes for the Gaffer cache service.
+- [Graph](graph.md): Contains the Gaffer `Graph` object and related utilities.
+- [Data](data.md): Classes defining Gaffer's data objects: `Element`, `Edge` and `Entity`.
+- [Store](store.md): Contains the Gaffer `Store` object and related classes.
+- [Serialisation](serialisation.md): Contains classes for Serialisation in Gaffer. 
+- [Accumulo Store](accumulo-store.md): A store implemented using Apache Accumulo.
+- [Core REST](core-rest.md): Classes which provide a Gaffer REST API using Jersey/JAX-RS.
+- [Spring REST](spring-rest.md): Implementation of the Gaffer REST API deployed in a Spring Boot container.
+
+## Project Architecture (Maven)
+
+Gaffer uses Maven and from this perspective the project is made up of multiple Maven modules. All components are Maven modules, but not all modules are components as some are just parent POMs (see below).
+Maven modules are not the same as Java modules (Java 9+), which Gaffer doesn't use or support. [See here for more info on multi-module builds and the Maven reactor](https://maven.apache.org/guides/mini/guide-multiple-modules.html).
+
+Gaffer's project structure involves three kinds of module/POM, mostly based on the [Maven packaging type](https://maven.apache.org/pom.html#Packaging). We can call these:
+
+- Parent modules (`pom`)
+- JAR modules (`jar`)
+- WAR only/Demo modules (`war`).
+
+Parent modules consist of a single POM file (parent POM) which defines various properties, dependencies and settings to be inherited by JAR modules or other parent modules.
```

Comment:
I'm fairly sure Gaffer is using aggregator poms in a few places as well rather than parents e.g. the 'core' module doesnt define any information that is inherited by its children its just an aggregator of other modules

Reply:
So the `core` POM is an aggregator but it is also inherited by everything in `core`. It is both an aggregator and a parent. There's actually no reason for it to be inherited as it doesn't set any properties, though this seems to be how things have been done in Gaffer.
It might be worth mentioning aggregator aspects as well, I was trying to keep the section brief but wasn't very successful at that anyway.